### DISCUSSION TITLE
How to do date validation

### DISCUSSION THREAD

**Discussion by za77** [CacheControl/json-rules-engine#278] [General]
Hi,
How to implement the date Validation ? 
How can we make the value attribute dynamic?

**Comment by tapaj** [comment]
Hi,

I was trying to implement the same thing, and I used `new Date("<dateString>").valueOf()` and used it with the existing operators.

**Comment by tapaj** [comment]
In order to make the value attribute inside your conditions dynamic use the `Fact()` with asynchronous call to load you fact and inside the condition you can spcify it as follows:

```
let rule = new Rule({
    conditions: {
        all: [{
            fact: "TEST_FACT_01",
            operator: "equal",
            value: {
                fact: "TEST_FACT_02",
            }
        }]
    }
}
```

`path` and `param` helpers are availble inside the rule for both `fact` and `value`.

Reference: https://github.com/CacheControl/json-rules-engine/issues/274#issue-949431157
